From 338dc32a634dd207e25d47a2f3615a59556c2b65 Mon Sep 17 00:00:00 2001
From: Jianxu Zheng <jian.xu.zheng@intel.com>
Date: Wed, 28 Jan 2015 11:48:58 +0800
Subject: [PATCH 264/269] atomisp2: adjust raw buffer delayed queue starting

there is a case that ov680 is stream on before ov8858 in depth mode.
this patch makes the delaying queue starting after all streams are
started and add condition for depth mode.

Change-Id: Ic8ca10444c33678438bf194af537e82e205832a3
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-19132
Signed-off-by: Jianxu Zheng <jian.xu.zheng@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/38043
Reviewed-by: Soren Friis <soren.friis@intel.com>
Integrator: Wei Tao <wei.tao@intel.com>
Approver: Wei Tao <wei.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 .../camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c    | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
index fc6a730..b4fc3da 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
@@ -1830,7 +1830,7 @@ start_sensor:
 			goto out;
 		}
 		__wdt_on_master_slave_sensor(isp, wdt_duration);
-		goto out;
+		goto start_delay_wq;
 	} else if (asd->depth_mode->val && (atomisp_streaming_count(isp) <
 		   ATOMISP_DEPTH_SENSOR_STREAMON_COUNT)) {
 		goto start_delay_wq;
-- 
1.9.1

